File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruriC/domain/Angajat.cs
namespace AgentiiZboruriC.domain;
using System;

[Serializable]
public class Angajat : Persoana
{
    internal string Username;
    internal string Password;

    public Angajat(int id, string nume, string prenume, string username, string password) : base(id, nume, prenume)
    {
        this.Username = username;
        this.Password = password;
    }

    public string GetUsername()
    {
        return Username;
    }

    public string GetPassword()
    {
        return Password;
    }

    public override string ToString()
    {
        return $"Angajat{{username='{Username}', password='{Password}'}}";
    }

    public int GetId()
    {
        return base.GetId();
    }

    public void SetId(int id)
    {
        base.SetId(id);
    }

    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;

        Angajat angajat = (Angajat)obj;

        return Username == angajat.Username && Password == angajat.Password;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Username, Password);
    }

    public void SetUsername(string username)
    {
        this.Username = username;
    }

    public void SetPassword(string password)
    {
        this.Password = password;
    }

    public void SetNume(string nume)
    {
        base.SetNume(nume);
    }

    public void SetPrenume(string prenume)
    {
        base.SetPrenume(prenume);
    }

    public string GetNume()
    {
        return base.GetNume();
    }

    public string GetPrenume()
    {
        return base.GetPrenume();
    }
}

File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruri/Model/src/main/java/ro/model/Angajat.java
package ro.model;

public class Angajat extends Persoana{
    private String username;
    private String password;

    public Angajat(Integer id, String nume, String prenume, String username, String password) {
        super(id, nume, prenume);
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    @Override
    public String toString() {
        return "Angajat{" +
                "username='" + username + '\'' +
                ", password='" + password + '\'' +
                '}';
    }

    @Override
    public Integer getId() {
        return super.getId();
    }

    @Override
    public void setId(Integer id) {
        super.setId(id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Angajat angajat = (Angajat) o;

        if (username != null ? !username.equals(angajat.username) : angajat.username != null) return false;
        return password != null ? password.equals(angajat.password) : angajat.password == null;
    }

    @Override
    public int hashCode() {
        int result = username != null ? username.hashCode() : 0;
        result = 31 * result + (password != null ? password.hashCode() : 0);
        return result;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void setNume(String nume) {
        super.setNume(nume);
    }

    public void setPrenume(String prenume) {
        super.setPrenume(prenume);
    }

    public String getNume() {
        return super.getNume();
    }

    public String getPrenume() {
        return super.getPrenume();
    }

}
